Bottom: 6a21a24c46b043933702d423a7d264a083d88020
Top:    8fe4a63f5769ca16e95a32c9dcc03f3be9ae6beb
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-01-24 02:00:54 +0000

Add configure checks for GtkGLEext, add to CFLAGS and LIBS

*** TODO: Need separate tests for libgl and libglut etc.. ***









---

diff --git a/configure.ac b/configure.ac
index 7c6aa38..18ab959 100644
--- a/configure.ac
+++ b/configure.ac
@@ -504,6 +504,24 @@ Please review the following errors:
 $GLIB_PKG_ERRORS])]
 	)
 	GLIB_VERSION=`$PKG_CONFIG glib-2.0 --modversion`
+
+
+	# Check for GtkGLExt
+	PKG_CHECK_MODULES(GTKGLEXT, gtkglext-1.0 >= 1.0.0, , [AC_MSG_ERROR([
+*** Required version of gtkglext is not installed - please install first ***
+Please review the following errors:
+$GTKGLEXT_PKG_ERRORS])]
+	)
+
+	GTKGLEXT_VER=`$PKG_CONFIG gtkglext-1.0 --modversion`
+
+	AC_CHECK_HEADERS(GL/glut.h)
+	case $ac_cv_header_GL_glut_h in
+	  no )
+	    AC_MSG_ERROR([You don't seem to have the GL glut library headers installed.])
+	    ;;
+	  * ) ;;
+	esac
 	;;
 
       nelma|png )
@@ -751,8 +769,8 @@ AM_CONDITIONAL(DEBUG_BUILD, test x$enable_debug = xyes)
 
 # ------------- Complete set of CFLAGS and LIBS -------------------
 
-CFLAGS="$CFLAGS $X_CFLAGS $DBUS_CFLAGS $GTK_CFLAGS"
-LIBS="$LIBS $XM_LIBS $DBUS_LIBS $X_LIBS $GTK_LIBS $DMALLOC_LIBS $GD_LIBS $INTLLIBS"
+CFLAGS="$CFLAGS $X_CFLAGS $DBUS_CFLAGS $GTK_CFLAGS $GTKGLEXT_CFLAGS"
+LIBS="$LIBS $XM_LIBS $DBUS_LIBS $X_LIBS $GTK_LIBS $GTKGLEXT_LIBS $DMALLOC_LIBS $GD_LIBS $INTLLIBS -lglut"
 
 
 # if we have gcc then add -Wall
